packages/flutter_goldens/lib/flutter_goldens.dart

```diff
@@ -69,222 +107,325 @@ abstract class FlutterGoldenFileComparator extends GoldenFileComparator {
+/// these tests are designed to be run post-submit on Cirrus CI, informed by the
+/// environment.
 ///
 /// See also:
 ///
 ///  * [GoldenFileComparator], the abstract class that
 ///    [FlutterGoldenFileComparator] implements.
-///  * [FlutterSkiaGoldFileComparator], another [FlutterGoldenFileComparator]
-///    that tests golden images through Skia Gold.
-class FlutterGoldensRepositoryFileComparator extends FlutterGoldenFileComparator {
-  /// Creates a [FlutterGoldensRepositoryFileComparator] that will test golden
-  /// file images against the `flutter/goldens` repository.
+///  * [FlutterPreSubmitFileComparator], another
+///    [FlutterGoldenFileComparator] that tests golden images before changes are
+///    merged into the master branch.
+///  * [FlutterLocalFileComparator], another
+///    [FlutterGoldenFileComparator] that tests golden images locally on your
+///    current machine.
+class FlutterSkiaGoldFileComparator extends FlutterGoldenFileComparator {
+  /// Creates a [FlutterSkiaGoldFileComparator] that will test golden file
+  /// images against Skia Gold.
   ///
   /// The [fs] and [platform] parameters useful in tests, where the default file
   /// system and platform can be replaced by mock instances.
-  FlutterGoldensRepositoryFileComparator(
-    Uri basedir, {
-    FileSystem fs = const LocalFileSystem(),
-    Platform platform = const LocalPlatform(),
+  FlutterSkiaGoldFileComparator(
+    final Uri basedir,
+    final SkiaGoldClient skiaClient, {
+    final FileSystem fs = const LocalFileSystem(),
+    final Platform platform = const LocalPlatform(),
   }) : super(
     basedir,
+    skiaClient,
     fs: fs,
     platform: platform,
   );
 
-  /// Creates a new [FlutterGoldensRespositoryFileComparator] that mirrors the
-  /// relative path resolution of the default [goldenFileComparator].
-  ///
-  /// By the time the future completes, the clone of the `flutter/goldens`
-  /// repository is guaranteed to be ready to use.
+  /// Creates a new [FlutterSkiaGoldFileComparator] that mirrors the relative
+  /// path resolution of the default [goldenFileComparator].
   ///
   /// The [goldens] and [defaultComparator] parameters are visible for testing
   /// purposes only.
-  static Future<FlutterGoldensRepositoryFileComparator> fromDefaultComparator({
-    GoldensRepositoryClient goldens,
+  static Future<FlutterSkiaGoldFileComparator> fromDefaultComparator(
+    final Platform platform, {
+    SkiaGoldClient goldens,
     LocalFileComparator defaultComparator,
   }) async {
-    defaultComparator ??= goldenFileComparator;
 
-    // Prepare the goldens repo.
-    goldens ??= GoldensRepositoryClient();
-    await goldens.prepare();
+    defaultComparator ??= goldenFileComparator;
+    final Directory baseDirectory = FlutterGoldenFileComparator.getBaseDirectory(
+      defaultComparator,
+      platform,
+    );
+    if (!baseDirectory.existsSync())
+      baseDirectory.createSync(recursive: true);
```

Comment:
Does stuff in the base directory ever get cleaned up?

Reply:
I do not think so.
For pre and post submit testing, the instance is fresh and clean each time. 
Locally, the test results/diffs are output there so I wouldn't want to dispose of them.

Reply:
For local testing: Let's say I run the tests and it outputs a diff for two tests. I fix the diff for one and run the tests again. Wouldn't it be confusing that the second diff is still there for the test I fixed? It feels like we need to clean this up at some point?